nomos-core/src/utils/asyncwritable.rs

```diff
@@ -0,0 +1,9 @@
+use std::{fmt::Debug, io, pin::Pin};
+
+use futures::AsyncWrite;
+
+/// Trait for types that can be written to an `AsyncWrite` stream.
+#[async_trait::async_trait]
+pub trait AsyncWritable: Debug {
+    async fn write_to(&self, writer: &mut Pin<Box<dyn AsyncWrite + Send>>) -> io::Result<()>;
+}
```

Comment:
I'm pretty sure this actually exists and we should use it instead.

Reply:
I wanted to define a trait for objects that can be written to the writer that implements `futures::AsyncWrite`. That is, this `write_to` function should define how the object can be serialized and written to the writer.
I’ll look for any existing trait similar as this.

Reply:
oh wait,, yeah, I completely missunderstood. It is actually the other way around. But anyway I feel something off here. I'll try to elaborate more with a suggested solution.